src/hnc_worker_sup.erl

```diff
@@ -17,39 +17,33 @@
 
 -behavior(supervisor).
 
--export([start_link/3]).
--export([start_worker/1, stop_worker/2]).
+-export([start_link/4]).
 -export([init/1]).
+-export([start_worker/1, stop_worker/2]).
 
--spec start_link(module(), term(), hnc:shutdown()) -> {ok, pid()}.
-start_link(Mod, Args, Shutdown) ->
-	supervisor:start_link(?MODULE, {Mod, Args, Shutdown}).
+start_link(Mod, Args, Shutdown, Modules) ->
+	supervisor:start_link(?MODULE, {Mod, Args, Shutdown, Modules}).
 
--spec start_worker(pid()) -> {ok, hnc:worker()}.
 start_worker(Sup) ->
 	supervisor:start_child(Sup, []).
 
--spec stop_worker(pid(), hnc:worker()) -> ok.
 stop_worker(Sup, Worker) ->
-	supervisor:terminate_child(Sup, Worker).
+	supervisor:terminate_child(Sup, Worker).	
 
-init({Mod, Args, Shutdown}) ->
-	{module, Mod}=code:ensure_loaded(Mod),
-	Modules=case erlang:function_exported(Mod, get_modules, 0) of
-		true -> Mod:get_modules();
-		false -> [Mod]
-	end,
+init({Mod, Args, Shutdown, Modules}) ->
+	ok=logger:update_process_metadata(#{hnc => #{module => ?MODULE}}),
 	{
 		ok,
 		{
 			#{
-				strategy => simple_one_for_one
+				strategy => simple_one_for_one,
```

Comment:
`simple_one_for_one`? You picked this strategy so we don't have to pass in `Mod`, `Args`, `Shutdown` and `Modules` into `start_worker`, right?

Reply:
That is correct.